engine/src/main/java/com/datatorrent/stram/codec/LogicalPlanSerializer.java

```diff
@@ -116,91 +165,144 @@ public boolean useForType(JavaType t)
-        String portName = portMeta.getPortName();
-        portDetailMap.put("name", portName);
-        portDetailMap.put("type", "output");
-        portDetailMap.put("attributes", portAttributeMap);
-        rawAttributes = Attribute.AttributeMap.AttributeInitializer.getAllAttributes(portMeta, Context.PortContext.class);
-        for (Map.Entry<Attribute<Object>, Object> attEntry : rawAttributes.entrySet()) {
-          portAttributeMap.put(attEntry.getKey().getSimpleName(), attEntry.getValue());
-        }
-        portList.add(portDetailMap);
+    }
+    return operators;
+  }
+
+  /**
+   * Return list of streams populated by module in top level dag.
+   * @param plan top level dag
+   * @param moduleMeta module
+   * @return
+   */
+  private static List<StreamMeta> getTopLevelStreams(LogicalPlan plan, ModuleMeta moduleMeta)
+  {
+    List<StreamMeta> streams = new ArrayList<StreamMeta>();
+    for (StreamMeta sMeta : moduleMeta.getDag().getAllStreams()) {
+      if (sMeta.getModuleName() == null) {
+        String fullName = moduleMeta.getFullName() + LogicalPlan.MODULE_NAMESPACE_SEPARATOR + sMeta.getName();
+        StreamMeta meta = plan.getStream(fullName);
+        streams.add(meta);
       }
     }
-    Collection<StreamMeta> allStreams = dag.getAllStreams();
+    return streams;
+  }
 
-    for (StreamMeta streamMeta : allStreams) {
-      HashMap<String, Object> streamDetailMap = new HashMap<String, Object>();
-      String streamName = streamMeta.getName();
-      streamMap.add(streamDetailMap);
-      String sourcePortName = streamMeta.getSource().getPortName();
-      OperatorMeta operatorMeta = streamMeta.getSource().getOperatorMeta();
-      HashMap<String, Object> sourcePortDetailMap = new HashMap<String, Object>();
-      sourcePortDetailMap.put("operatorName", operatorMeta.getName());
-      sourcePortDetailMap.put("portName", sourcePortName);
-      streamDetailMap.put("name", streamName);
-      streamDetailMap.put("source", sourcePortDetailMap);
-      List<InputPortMeta> sinks = streamMeta.getSinks();
-      ArrayList<HashMap<String, Object>> sinkPortList = new ArrayList<HashMap<String, Object>>();
-      for (InputPortMeta sinkPort : sinks) {
-        HashMap<String, Object> sinkPortDetailMap = new HashMap<String, Object>();
-        sinkPortDetailMap.put("operatorName", sinkPort.getOperatorWrapper().getName());
-        sinkPortDetailMap.put("portName", sinkPort.getPortName());
-        sinkPortList.add(sinkPortDetailMap);
+  private static List<Map<String, Object>> getLogicalModulesInfo(LogicalPlan dag, boolean flatten)
+  {
+    List<Map<String, Object>> modulesArray = new ArrayList<Map<String, Object>>();
+    for (ModuleMeta moduleMeta : dag.getAllModules()) {
+      modulesArray.add(fillLogicalModuleDetails(dag, moduleMeta, flatten));
+    }
+    return modulesArray;
+  }
+
+  private static Map<String, Object> fillLogicalModuleDetails(LogicalPlan dag, ModuleMeta moduleMeta, boolean flatten)
```

Comment:
Looks much like StreamingContainerManager.getLogicalModuleInfo. This may be a pre-existing pattern, but why? Maybe @davidyan74 can look at this.